### ISSUE TITLE
Feature Request: Generate Fixdat reports for red items only.

### ISSUE THREAD

**Issue by LnmVolbo** [gjefferyes/RomVault#36]
Currently the "Generate Reports" button always lists both the red (completely missing) and yellow (can be fixed) items.

Recently a request was made on RS: **make fixdat for an unbuild set?**: http://romshepherd.com/index.php/topic,5230.0.html

It boils down to omitting the yellow colored items when generating the fixdat reports, and only reporting the completely missing items. (Which means _not_ reporting items which are not fixed, but could be)

This can be used to "subtract" the items listed in a .dat file from collected sets, The requester wants to find what is missing from a collection without having to rebuild it first.

It appears that line 115 in _Report.cs_ controls which items get listed in the fixdat reports.

If one changes the line from
```cs
if (tRom.DatStatus == DatStatus.InDatCollect && tRom.GotStatus != GotStatus.Got)
```
to:
```cs
if (tRom.DatStatus == DatStatus.InDatCollect && tRom.GotStatus != GotStatus.Got && !(tRom.RepStatus == RepStatus.CanBeFixed || tRom.RepStatus == RepStatus.CorruptCanBeFixed))
```
the requested behavior can be accomplished.

If this change is approved and made part of the official release, users should be cautioned that only red items will be reported, formerly both red and yellow items were.

If a user inadvertently uses the "Find Fixes" button immediately before the "Generate Reports" button, it would be possible for some items not to be listed in the fixdat files.  To circumvent this, users that wish to always see all the missings in the reports should make a habit of pressing the "Scan ROMs" button (and therefore clearing the found items) just before using the "Generate Reports" button.

Another way to handle this is to have a preference setting added to the "Settings" dialog that forces the behavior (either red and yellow, or red only).

**Comment by LnmVolbo** [comment]
Feedback from the RS forum: http://romshepherd.com/index.php/topic,5271.msg64218.html#msg64218

...indicates that the resulting fixdat files have "empty" (of rom or disk info) game contents for the places that were colored yellow when the fixdat was created.

This may be undesirable clutter, although it might be useful for keeping track of what games haven't been fixed yet.

As a result, I have made it the default behavior to remove the clutter and optional to leave these game "skeletons" in the fixdat files.  This can be accomplished by clicking "Generate Reports" with either the left, or some other button.  The left button results in fixdat files with the clutter removed, and any other button (middle, right, etc...) will give the fixdat files with the clutter included.  I have added a tooltip on the "Generate Reports" button to remind the user.  And if Yellow items were excluded, a comment will appear in the header of the fixdat file as a reminder.

The clutter is removed by a new routine _scrubIt()_.  This routine is executed immediately after each fixdat file is created. The fixdat file is re-read and only games that have valid rom or disk contents are kept.  If none of the games have contents, that particular fixdat file is deleted.  If there are games in the fixdat with valid contents, the fixdat is deleted and the new version is saved.  If there are any Yellow items removed from the fixdat (tracked by comparing _rpy and _ro), a comment is inserted into the header of the revised fixdat file before it is saved.  The comment includes the number of items not appearing in the fixdat - a useful number for those who are measuring how much collections overlap.

This method of creating the .dat files, then revising them afterwards, works well when there are a small to medium amount of Yellow items to remove.  If one were doing a series of large collection comparisons, with a large amount of Yellow items, there is a possibility of a performance issue.  But that remains to be seen.

This more detailed implementation is a bit more complicated than the one liner indicated in the first post of this issue.  Attached are the changed _Reports.cs_ and _FrmMain.cs_ sources.  Since I added a tooltip and rewired the "Generate Reports" button, changes were also made in the _FrmMain.Designer.cs_ file.  It is attached for reference, but it may not be wise to overwrite your existing _FrmMain.Designer.cs_ file, as VS wants to generate it itself.  Here is a list of changes and additions made in the form designer for FrmMain:

Removed the _BtnReportClick()_ routine and replaced it with a _BtnReport_MouseUp()_ routine.
Added a tooltip control with the default name.
Added at tooltip string to btnReport with the following (two line) contents:
- Left button = Red items only
- Other buttons = Red + Yellow items

[Sources for #36.zip](https://github.com/gjefferyes/RomVault/files/298090/Sources.for.36.zip) (20 KB)

I'm not sure if **GordonJ** wants his creation covered with tooltips, so I understand if this is over-implemented and not all of these changes make it into an official version.

Also linked is a revised windows binary, named ROMVault226x2.  As well as these changes to #36, it also includes the changes made from issue #38.  I'm still seeing the issue with the .config file, so that is included in the .zip file.

https://www.sendspace.com/file/7ukmzh (461 KB)

Summary of changes for the .exe in the attached .zip file:

- Everything in first version of #36, plus #38 
- Default creation of clean "Red items" only fixdats, option to create cluttered version.
- Number of Yellow items excluded from "Red items" only fixdat appears in a comment in the header of the fixdat - useful for tabulating overlaps between collections.
